tornado/web.py

```diff
@@ -2809,6 +2809,23 @@ def validate_absolute_path(self, root: str, absolute_path: str) -> Optional[str]
                 self.redirect(self.request.path + "/", permanent=True)
                 return None
             absolute_path = os.path.join(absolute_path, self.default_filename)
+        else:
+            canonical_path = os.path.normcase(os.path.normpath(absolute_path))
```

Comment:
I don't think `os.path.normcase()` does the job here. It's an identity function on macOS, even though macOS filesystems are case-insensitive by default (and I think even the windows version does the wrong thing if the system locale is set to Turkish - `normcase` uses the "invariant" locale, but the actual windows filesystem is sensitive to locale changes). I'm not sure what, if anything, `normcase` is good for since the actual rules are AFAIK determined on a per-volume basis instead of system-wide.

Reply:
Might be true, I have not tried this stuff on macOS or Windows.